### ISSUE TITLE
how to convert python tf.get_variable to js

### ISSUE THREAD

**Issue by aihuawu** [tensorflow/tfjs#1547]
python
  embedding_table = tf.get_variable(
      name=word_embedding_name,
      shape=[vocab_size, embedding_size],
      initializer=create_initializer(initializer_range))

how can i convert it to js?
Thanks.

george wu

**Comment by caisq** [comment]
In Python `tf.get_variable()` is used to create variables. In TensorFlow.js, you can create variables by using `tf.variable()`. See documentation at: https://js.tensorflow.org/api/latest/#variable 

Let me know if that answers your question.

**Comment by caisq** [comment]
However, be aware that `tf.variable()` in TensorFlow.js does not support initializers. You have to provide an explicit initial value.

**Comment by aihuawu** [comment]
Thank you.
Still some question on this:
1) If I use the same variable name, and call it twice, I should get the same variable. Is this true for js version?
2) how can I use the equivalent js version for the python version:
with tf.variable_scope("encoder")

George Wu

**Comment by caisq** [comment]
@aihuawu 


1: No. tf.variable() in TensorFlow.js doesn't support this reusing pattern. If you want to use a variable twice, just use the same variable object.
2. There is no `tf.variable_scope()` in TensorFlow.js. `tf.variable_scope()` has been deprecated in TensorFlow 2.0 also. Users are now recommended to achieve Layer/Variable reusing through keras layers. TF.js has an equivalent of keras API that helps you do that. If you can be more specific on what you are trying to achieve, I may be able to help further.

**Comment by aihuawu** [comment]
Thank you.
So you recommend me to just use Layer which has member of Variable. And then I just keep the Layer reference and reuse this layer in other location.  I will try that.

**Comment by aihuawu** [comment]
@caisq 

For python:
  tvars = tf.trainable_variables()
  grads = tf.gradients(loss, tvars)

I have found js:
tf.variableGrads (f, varList?) 

Is it possible to get all trainable_variables in js? 
Or should I keep reference of all the trainable_variables myself?

Thanks.

George wu

**Comment by caisq** [comment]
`tf.variableGrads()` calculates gradients for all trainable variables by default, if you don't specify the 2nd argument.

**Comment by aihuawu** [comment]
Great, Thanks.
That's all I need so far.

George Wu
